### DISCUSSION TITLE
How to get a response from the server after each chunk?

### DISCUSSION THREAD

**Discussion by SvistelnykOleksandr** [rpldy/react-uploady#510] [Q&A] [answered]
I use ChunkedUploady and hook useItemProgressListener. Server response such json: {done: 7, status: true}, but object that return hook useItemProgressListener hasn't uploadResponse. What could be the problem?
Thanks for attention, I will be glad if you can help

**Comment by yoavniran** [comment, marked as answer]
Hey @SvistelnykOleksandr,

The progress listener doesnt have information related to chunks. Its only getting the overall progress info.

For the chunk upload response you should use the [useChunkFinishListener](https://react-uploady.org/docs/api/hooks/useChunkFinishListener/) hook.

It receives an uploadData object with a response property that should contain the server response.

```jsx

import { useChunkFinishListener } from "@rpldy/chunked-uploady";

const MyComponent = () => {
    useChunkFinishListener(({ uploadData }) => {
        console.log(`response data = `, uploadData.response);
    });
};

```

**Reply by SvistelnykOleksandr** [reply]
Thank you very much!!!
